ngraph/core/include/ngraph/op/max_pool.hpp

```diff
@@ -76,21 +56,85 @@ namespace ngraph
                               const HostTensorVector& inputs) const override;
                 bool has_evaluate() const override;
 
-            protected:
-                Shape m_kernel;
-                Strides m_strides;
-                Shape m_pads_begin;
-                Shape m_pads_end;
-                PadType m_auto_pad;
-                op::RoundingType m_rounding_type;
-
             private:
-                bool update_auto_padding(const PartialShape& in_shape,
-                                         Shape& new_pads_end,
-                                         Shape& new_pads_begin) const;
                 bool evaluate_maxpool(const HostTensorVector& outputs,
                                       const HostTensorVector& inputs) const;
             };
         } // namespace v1
+
+        namespace v8
+        {
+            /// \brief MaxPooling operation with values and indices calculated as individual outputs
+            class NGRAPH_API MaxPool : public op::util::MaxPoolBase
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+
+                /// \brief Constructs an empty MaxPool operation.
+                MaxPool() = default;
+
+                /// \brief Constructs a parametrized MaxPool operation.
+                ///
+                /// \param arg Output of a node producing the feature tensor to be pooled.
+                /// \param strides The strides of the pooling filter.
+                /// \param dilations The dilations of the pooling filter.
+                /// \param pads_begin Paddings at the beginning of each spatial axis.
+                /// \param pads_end Paddings at the end of each spatial axis.
+                /// \param kernel The kernel shape.
+                /// \param rounding_type Whether to use ceiling or floor rounding type while
+                ///                      computing the output shape.
+                /// \param auto_pad The pad type for automatic calculation of the padding sizes.
+                /// \param index_element_type The data type used by the second output tensor
+                ///                           containing the selected indices.
+                /// \param axis Indicates a dimension in the input data shape which should be used
+                ///             as a starting point for calculation of the upper bound of allowed
+                ///             values of the indices output.
+                MaxPool(const Output<Node>& arg,
+                        const Strides& strides,
+                        const Strides& dilations,
+                        const Shape& pads_begin,
+                        const Shape& pads_end,
+                        const Shape& kernel,
+                        const op::RoundingType rounding_type = op::RoundingType::FLOOR,
+                        const PadType& auto_pad = op::PadType::EXPLICIT,
```

Comment:
Why do we have a mix of passing enumerations by reference and by value?

Reply:
Right, there were inconsistencies and I just copy and pasted them. Fixed.